components/ValueTypeDecoder/ValueTypeDecoder.tsx

```diff
@@ -1,77 +1,130 @@
-    return (
-      <>
-        <code>{value}</code>
-        <AddressButtons address={value} />
-      </>
-    );
-  }
+const ValueTypeDecoder: React.FC<Props> = ({
+  provider,
+  address,
+  erc725JSONSchema,
+  value,
+}) => {
+  const [decodedDataOneKey, setDecodedDataOneKey] = useState<{
+    [key: string]: any;
+  }>([]);
+  const [fetchedData, setFetchedData] = useState<DecodeDataOutput>({
+    key: '',
+    name: '',
+    value: [],
+  });
 
-  // The schema may be wrong, this error will be catched bellow
-  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  // @ts-ignore
-  const schema: ERC725JSONSchema[] = [erc725JSONSchema];
+  useEffect(() => {
+    const startDecoding = async () => {
+      if (address) {
+        const schema: ERC725JSONSchema[] = [erc725JSONSchema];
+        const erc725 = new ERC725(schema, address, provider.currentProvider);
 
-  const erc725 = new ERC725(schema);
+        const decodedData = erc725.decodeData([
+          {
+            keyName: erc725JSONSchema.name,
+            value: value as string,
+          },
+        ]);
 
-  let decodedDataOneKey;
-  try {
-    decodedDataOneKey = erc725.decodeData({
-      [erc725JSONSchema.name]: value,
-    });
-  } catch (err) {
-    // Goes here if key is not in erc725.js yet or if key is undefined
-    return <span>Can&apos;t decode this key</span>;
-  }
+        setDecodedDataOneKey(decodedData);
+
+        const result = await erc725.getData(erc725JSONSchema.name);
+        console.log(result);
+        setFetchedData(result);
+      }
+    };
+    startDecoding();
+  }, [address]);
 
   try {
-    return (
-      <div>
-        <pre>
-          {JSON.stringify(decodedDataOneKey[erc725JSONSchema.name], null, 4)}
-        </pre>
-        {decodedDataOneKey[erc725JSONSchema.name].url && (
+    if (typeof decodedDataOneKey[0].value === 'string') {
+      if (erc725JSONSchema.valueContent === 'Address') {
+        return <AddressView value={decodedDataOneKey[0].value} />;
+      }
+
+      return <code>{value}</code>;
+    }
+
+    if (
+      fetchedData !== undefined &&
+      Array.isArray(fetchedData.value as any) &&
+      erc725JSONSchema.keyType === 'Array'
+    ) {
+      return (
+        <ul>
+          {(fetchedData.value as string[]).map((item, index) => (
+            <li key={index}>
+              <code>{item}</code>
+            </li>
+          ))}
+        </ul>
+      );
+    }
+
+    if (erc725JSONSchema.valueContent === 'JSONURL') {
+      return (
+        <>
+          <pre>{JSON.stringify(decodedDataOneKey[0].value, null, 4)}</pre>
           <div>
-            <span>URL: {decodedDataOneKey[erc725JSONSchema.name].url}</span> -{' '}
-            {decodedDataOneKey[erc725JSONSchema.name].url.indexOf('ipfs://') !==
-              -1 && (
+            <span>URL: {decodedDataOneKey[0].value.url}</span> -{' '}
+            {decodedDataOneKey[0].value.url.indexOf('ipfs://') !== -1 && (
```

Comment:
not sure why i added this ipfs:// check - is it relevant?

Reply:
The IPFS hash of the content would be enough imo. But still relevant I would say.